R/foverlaps.R

```diff
@@ -128,24 +129,15 @@ foverlaps <- function(x, y, by.x = if (!is.null(key(x))) key(x) else key(y), by.
   }
   # nomatch has no effect here, just for passing arguments consistently to `bmerge`
   .Call(Clookup, uy, nrow(y), indices(uy, y, yintervals, nomatch=0L, roll=roll), maxgap, minoverlap, mult, type, verbose)
-  if (maxgap == 0L && minoverlap == 1L) {
```

Comment:
@arunsrinivasan check out this diff. axed `iintervals` as it appears unused. And axed the extra argument checks to simplify so long as the other options are unimplemented

Reply:
@MichaelChirico good catch regarding `iintervals`. Seems like it wasn't used right from the first commit of `foverlaps`.

Reply:
I understand why you simplified code, but I left it there to serve as a placeholder when I'll come back to work on it... The first `if` statement would always be run currently. So that's the only cost here, which is fine.. Would it be possible to revert this part to how it was please?

The PR overall looks great! Apologies for not being able to write earlier.

Reply:
@arunsrinivasan I figured that's why you left it like that... only changed the logic to make testing easier, Codecov was complaining since it's impossible to get to the other branches to test them (all hail the might Codecov 🌞  ⛪️ ). I figured it'd be easy enough to revert to the old code when needed (that's what git's for right?)

Reply:
Why not wrap those lines with `# nocov start` and `# nocov end` so as to skip coverage tests? See https://github.com/Rdatatable/data.table/blob/70db3e48738026388c6487e83ee902ad37256ca5/R/c.factor.R

Yes, commented out with a comment pointing to this PR would be okay too.

Reply:
simply because i didn't know that was possible! I'll add that to `onAttach` and `onLoad` as well... thanks!